client/state/partner-portal/partner/selectors.ts

```diff
@@ -63,3 +63,10 @@ export function hasValidPaymentMethod( state: PartnerPortalStore ): boolean {
 	const partner = getCurrentPartner( state );
 	return partner?.has_valid_payment_method || false;
 }
+
+export function isPaymentMethodRequired( state: PartnerPortalStore ): boolean {
```

Comment:
Could the name potentially be misunderstood? E.g.:

1. Do we check if a specific **partner type** requires a payment method?
2. Do we check if a specific partner already has a payment method?
3. Do we check if a specific action requires a registered payment method?

Would renaming it to something like `doPartnerRequirePaymentMethod` make it more specific?